File: Data Structures & Algorithms/is-palindrome/submission-0.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        st = s.lower()
        st = ''.join(char for char in st if char.isalnum())

        left = 0
        right = len(st) -1

        while left < right:
            if st[left] != st[right]:
                return False
            left += 1
            right -= 1
        
        return True


        

File: Data Structures & Algorithms/max-water-container/submission-0.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let ans = 0, left = 0, right = heights.length-1;

        while(left < right){
            let width = right - left
            let curr_height = Math.min(heights[left], heights[right])

            let max_area = curr_height * width
            ans = Math.max(ans, max_area)

            if(heights[left] < heights[right]){
                left++
            } else {
                right--
            }
        }

        return ans
    }
}
